mitodl/mit-open
# Config to set cache control headers

### What are the relevant tickets?

https://github.com/mitodl/hq/issues/5796


### Description (What does it do?)


Next.js sets `no-cache` for any dynamically rendered pages, though we have taken the approach of ensuring all server rendered HTML is public and can be stored on shared caches.

This change overrides this behavior by setting cache headers explicitly for all of our routes.

### How can this be tested?

_Must be tested in Production mode._

Locally you should see `Cache-Control: public, max-age=120` on all base HTML responses, but not on JS chunks, images or other static content.

Once deployed you should see CDN cache hits on subsequent requests (X-Cache header).

### Additional Context



- This also upgrades Next.js to 14.2.15 so that we have this recent change in [v14.2.10](https://github.com/vercel/next.js/releases/tag/v14.2.10) https://github.com/vercel/next.js/pull/69802.

- Several routes are static rendered at build time (/about, /departments, /topics, /units, etc). Next.js gives these an `s-maxage=31536000`, though we're setting all to the same for consistency.

- This does also set the cache header on RSC payloads e.g. /?resource=6135&_rsc=1wtp7. This may cause issues.

- This only sets a short TTL of 120s - wanting to be cautious while we first test.
